packages/manager/apps/pci-block-storage/src/pages/retype/Retype.page.tsx

```diff
@@ -0,0 +1,143 @@
+  const navigate = useNavigate();
+  const { projectId, volumeId } = useParams();
+  const {
+    data,
+    preselectedEncryptionType,
+    isPending,
+  } = useCatalogWithPreselection(projectId, volumeId);
+  const onClose = () => navigate('..');
+
+  const [newVolumeModel, setNewVolumeModel] = useState<TVolumeModel>(undefined);
+
+  const [
+    newEncryptionType,
+    setNewEncryptionType,
+  ] = useState<EncryptionType | null>(preselectedEncryptionType);
+
+  const currentType = useMemo(() => data?.find((m) => m.isPreselected), [
+    data?.length,
+  ]);
+
+  useEffect(() => {
+    setNewVolumeModel(currentType);
+  }, [currentType]);
+
+  useEffect(() => {
+    setNewEncryptionType(preselectedEncryptionType);
+  }, [preselectedEncryptionType]);
+
+  const nothingHasChanged = useMemo(
+    () =>
+      currentType?.name === newVolumeModel?.name &&
+      preselectedEncryptionType === newEncryptionType,
+    [currentType, newVolumeModel, preselectedEncryptionType, newEncryptionType],
+  );
+
+  const displayedData =
+    data?.length > 0 ? (
+      <div>
+        <legend>
+          <Subtitle>
+            {t('retype:pci_projects_project_storages_blocks_retype_title')}
+          </Subtitle>
+        </legend>
+        <OsdsText size={ODS_TEXT_SIZE._400} level={ODS_TEXT_LEVEL.subheading}>
+          {t('retype:pci_projects_project_storages_blocks_retype_subtitle')}
+        </OsdsText>
+        <VolumeTypeTilesInput
+          value={newVolumeModel}
+          data={data}
+          onChange={(e) => setNewVolumeModel(e)}
+          label={t(
+            'retype:pci_projects_project_storages_blocks_retype_change_type_title',
+          )}
+        />
+        <div>
+          <Encryption
+            encryptionType={newEncryptionType}
+            onChange={(e) => setNewEncryptionType(e)}
+            texts={{
+              title: t(
+                'retype:pci_projects_project_storages_blocks_retype_change_encryption_title',
+              ),
+            }}
+          />
+        </div>
+      </div>
+    ) : (
+      <OsdsMessage
+        type={ODS_MESSAGE_TYPE.warning}
+        data-testid="retype-cant-retype"
+      >
+        {t('retype:pci_projects_project_storages_blocks_retype_cant_retype')}
+      </OsdsMessage>
+    );
+
+  return (
+    <OsdsModal onOdsModalClose={onClose}>
+      <slot>
+        {!isPending ? (
+          displayedData
+        ) : (
+          <OsdsSpinner
+            size={ODS_SPINNER_SIZE.md}
+            inline
+            className="mt-12 flex justify-center"
+            data-testid="retypePage-spinner"
+          />
+        )}
+      </slot>
+      <OsdsButton
+        slot="actions"
+        color={ODS_THEME_COLOR_INTENT.primary}
+        variant={ODS_BUTTON_VARIANT.ghost}
+        onClick={onClose}
+      >
+        {t('common:pci_projects_project_storages_blocks_button_cancel')}
+      </OsdsButton>
+      <OsdsButton
+        slot="actions"
+        color={
+          nothingHasChanged
```

Comment:
I think the style should be handled by ODS directly whenever it is disabled or not